corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -256,21 +253,8 @@ def _update_translation(self, row, language_dict, require_translation=True):
             ))
 
     def _update_id_mappings(self, rows, detail, langs=None):
-        if len(rows) == len(detail.enum) or not toggles.ICDS.enabled(self.app.domain):
-            for row, mapping in zip(rows, detail.enum):
-                self._update_translation(row, mapping.value)
-        else:
-            # Not all of the id mappings are described.
-            # If we can identify by key, we can proceed.
-            mappings_by_prop = {mapping.key: mapping for mapping in detail.enum}
-            if len(detail.enum) != len(mappings_by_prop):
-                self.msgs.append((
-                    messages.error,
-                    _("You must provide all ID mappings for property '{}'").format(detail.field)))
-            else:
-                for row in rows:
-                    if row['id'] in mappings_by_prop:
-                        self._update_translation(row, mappings_by_prop[row['id']].value)
+        for row, mapping in zip(rows, detail.enum):
```

Comment:
isn't the else block relevant `if len(rows) != len(detail.enum)` ?

Reply:
> Isn't that not possible?

I actually don't know.

Reply:
All this time I had assumed you wrote this code, since you've done so much in translations. It was @esoergel  - Ethan, can you weigh in?

Reply:
I had no recollection of this, so I dug up my [original commit](https://github.com/dimagi/commcare-hq/commit/9a4ab1c4f2af54c311ed776d60086a2b794f1faf).  Looks like I was pretty unhappy with the state of things:
> This whole bulk upload code is pretty ugly and in dire need of a
refactor.  The state model is spread all over the place, it relies
heavily on mutating dictionaries and modifying variables that are part
of a very broad scope.

From the [PR description](https://github.com/dimagi/commcare-hq/pull/22106), it looks like the point of this was to allow uploading only some ID mappings, without specifying all of them.  It was intended only to be supported for ICDS, and I agree with Jenny's interpretation here that since we can assume `toggles.ICDS` isn't enabled, the else branch can be deleted.

It does seem like we should throw an explicit error if `len(rows) != len(detail.enum)` to avoid weird behavior, but I can see that as outside of the scope of this PR, which is only removing an unused feature.

Reply:
> It does seem like we should throw an explicit error if len(rows) != len(detail.enum) to avoid weird behavior, but I can see that as outside of the scope of this PR, which is only removing an unused feature.

Sounds like worth having this as a part of this PR itself. I assume that a partial upload was permitted as a side effect of that "or" condition, even for non-ICDS enabled flag and would dangerously update translations. So avoiding that would be nice.

Reply:
I'm going to be strict about the scope on this one, would prefer not to spend any more time on it. Would either of you mind throwing me an approval, since we agree the current code is fine?